### DISCUSSION TITLE
Any way to get last location from the history stack?

### DISCUSSION THREAD

**Discussion by StelFoog** [TanStack/router#305] [Q&A] [answered]
As the title says I'm trying to figure out if there's any way to see what the last location was. Or if that's not possible, if there's a way to pass some data that's not included in the URL included based on which link to the view is used.

**Comment by tannerlinsley** [comment, marked as answer]
Not yet. But there will be in the future.

**Reply by StelFoog** [reply]
Is there a rough timeline for when this will be added?

**Reply by Jared-Dahlke** [reply]
@StelFoog this would be required to redirect the user back to whatever page they were trying to go to after they login right?

so for example this would be in Login.js, and we have no way to get the 'referer' url correct? or do you know of a way using window.location?
 
if(loggedIn){
  <Navigate to={referer}
}

**Reply by jonandgon** [reply]
Any movement on this issue? It would be great. Right now I'm using session storage to load route history which isn't ideal...

**Reply by jonandgon** [reply]
Using your [kitchen sink single file demo](https://tanstack.com/router/v1/docs/examples/react/kitchen-sink-single-file?file=src%2Fmain.tsx) as an example of desired functionality:
- Go to `/dashboard` and click on "invoices"
- Go to some other route, such as `/expensive`
- Go back to `/dashboard`
  - I would like to be able to go back to invoices automatically without having to click on invoices again.

**Reply by ali-issa** [reply]
Hey @jonandgon, did you find a cleaner alternative to using session storage?

**Reply by schiller-manuel** [reply]
session storage is the way to go here, otherwise this would not support reloads. a session storage based solution might be on its way via a user contribution.

**Comment by dotintegral** [comment]
Hey. Is there an update to this request? Known timeline?

**Comment by 1057105012** [comment]
I don't think this is the responsibility of routing, or that this requirement is actually dispensable. You can use some useHistoryContext hooks like this to handle it. He is not complicated
